### PR TITLE
[4.0] [com_content] [wip] convert to prepared statement

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes
use prepared statement for SQL

### Testing Instructions

test com_content

### Expected result

should work as before

### Actual result
N/A

### CODE REVIEW

File: administrator/components/com_content/Helper/ContentHelper.php [hunk 1]
```diff
@@ -175,9 +181,11 @@ public static function onPrepareForm(Form $form, $data)
 					}
 					elseif ((int) $workflow_id > 0)
 					{
-					$query	->clear('where')
-								->where($db->quoteName('id') . ' = ' . (int) $workflow_id)
-								->where($db->quoteName('published') . ' = 1');
+						$workflowId = (int) $workflow_id;
+						$query->clear('where')
+								->where($db->quoteName('id') . ' = :worflowid')
```
Comment (Quy) [inline administrator/components/com_content/Helper/ContentHelper.php]:
```suggestion
								->where($db->quoteName('id') . ' = :workflowid')
```

File: administrator/components/com_content/Helper/ContentHelper.php [hunk 2]
```diff
@@ -175,9 +181,11 @@ public static function onPrepareForm(Form $form, $data)
...
+								->where($db->quoteName('published') . ' = 1')
+								->bind(':worflowid', $workflowId, ParameterType::INTEGER);
```
Comment (Quy) [inline administrator/components/com_content/Helper/ContentHelper.php]:
```suggestion
								->bind(':workflowid', $workflowId, ParameterType::INTEGER);
```

File: administrator/components/com_content/Model/ArticleModel.php [hunk 3]
```diff
@@ -1028,7 +1035,7 @@ public function featured($pks, $value = 0)
 				$query = $db->getQuery(true)
 					->select('f.content_id')
```
Comment (Quy) [inline administrator/components/com_content/Model/ArticleModel.php]:
```suggestion
					->select($db->quoteName('f.content_id'))
```

File: administrator/components/com_content/Model/ArticleModel.php [hunk 4]
```diff
@@ -1028,7 +1035,7 @@ public function featured($pks, $value = 0)
...
 					->from('#__content_frontpage AS f')
```
Comment (Quy) [inline administrator/components/com_content/Model/ArticleModel.php]:
```suggestion
					->from($db->quoteName('#__content_frontpage ', 'f'))
```

File: administrator/components/com_content/Model/ArticlesModel.php [hunk 5]
```diff
@@ -319,37 +320,42 @@ protected function getListQuery()
...
 
 		// Filter by published state
 		$workflowStage = (string) $this->getState('filter.stage');
 
 		if (is_numeric($workflowStage))
 		{
-			$query->where('wa.stage_id = ' . (int) $workflowStage);
+			$workflowStage = (int) $workflowStage;
+			$query->where($db->quoteName('wa.stage_id') . ' = :workflowstage')
+				->bind('workflowstage', $workflowStage, ParameterType::INTEGER);
```
Comment (Quy) [inline administrator/components/com_content/Model/ArticlesModel.php]:
```suggestion
				->bind(':workflowstage', $workflowStage, ParameterType::INTEGER);
```

File: administrator/components/com_content/Model/ArticlesModel.php [hunk 6]
```diff
@@ -430,29 +441,40 @@ protected function getListQuery()
...
+				$query->where('(' . $db->quoteName('ua.name') . ' LIKE :name' .
+					' OR ' . $db->quoteName('ua.username') . ' LIKE :uname'. ')');
+				$query->bind(':name', $search)
+					->bind(':uname', $search);
 			}
 			elseif (stripos($search, 'content:') === 0)
 			{
-				$search = $db->quote('%' . $db->escape(substr($search, 8), true) . '%');
+				$search = '%' . substr($search, 8) . '%';
 				$query->where('(a.introtext LIKE ' . $search . ' OR a.fulltext LIKE ' . $search . ')');
```
Comment (Quy) [inline administrator/components/com_content/Model/ArticlesModel.php]: Convert.

File: administrator/components/com_content/Model/FeaturedModel.php [hunk 7]
```diff
@@ -249,29 +258,40 @@ protected function getListQuery()
...
+				$query->where('(' . $db->quoteName('ua.name') . ' LIKE :name' .
+					' OR ' . $db->quoteName('ua.username') . ' LIKE :uname'. ')');
+				$query->bind(':name', $search)
+					->bind(':uname', $search);
 			}
 			elseif (stripos($search, 'content:') === 0)
 			{
-				$search = $db->quote('%' . $db->escape(substr($search, 8), true) . '%');
+				$search = '%' . substr($search, 8) . '%';
 				$query->where('(a.introtext LIKE ' . $search . ' OR a.fulltext LIKE ' . $search . ')');
```
Comment (Quy) [inline administrator/components/com_content/Model/FeaturedModel.php]: Convert.

File: administrator/components/com_content/Model/FeaturedModel.php [hunk 8]
```diff
@@ -284,14 +304,14 @@ protected function getListQuery()
 
 		if (is_array($tagId))
 		{
-			$tagId = implode(',', ArrayHelper::toInteger($tagId));
+			$tagId = ArrayHelper::toInteger($tagId);
 
 			if ($tagId)
 			{
 				$subQuery = $db->getQuery(true)
 					->select('DISTINCT content_item_id')
```
Comment (Quy) [inline administrator/components/com_content/Model/FeaturedModel.php]: Convert

File: administrator/components/com_content/Model/FeaturedModel.php [hunk 9]
```diff
@@ -284,14 +304,14 @@ protected function getListQuery()
...
 					->from($db->quoteName('#__contentitem_tag_map'))
-					->where('tag_id IN (' . $tagId . ')')
+					->whereIn($db->quoteName('tag_id'), $tagId)
 					->where('type_alias = ' . $db->quote('com_content.article'));
```
Comment (Quy) [inline administrator/components/com_content/Model/FeaturedModel.php]: Convert

File: administrator/components/com_content/Model/FeaturedModel.php [hunk 10]
```diff
@@ -284,14 +304,14 @@ protected function getListQuery()
...
 
 				$query->join('INNER', '(' . (string) $subQuery . ') AS tagmap ON tagmap.content_item_id = a.id');
```
Comment (Quy) [inline administrator/components/com_content/Model/FeaturedModel.php]: Convert innerJoin

File: administrator/components/com_content/Service/HTML/AdministratorService.php [hunk 11]
```diff
@@ -50,6 +51,7 @@ public function association($articleid)
 			}
 
 			// Get the associated menu items
+			$associations = array_values($associations);
 			$db = Factory::getDbo();
 			$query = $db->getQuery(true)
 				->select('c.*')
```
Comment (Quy) [inline administrator/components/com_content/Service/HTML/AdministratorService.php:57]: Convert 57-67.

File: components/com_content/Model/ArticleModel.php [hunk 12]
```diff
@@ -171,13 +181,13 @@ public function getItem($pk = null)
 				{
 					throw new \Exception(Text::_('COM_CONTENT_ERROR_ARTICLE_NOT_FOUND'), 404);
 				}
-
+/*
```
Comment (Quy) [inline components/com_content/Model/ArticleModel.php:184]: Comment??

File: components/com_content/Model/ArticlesModel.php [hunk 13]
```diff
@@ -297,21 +297,24 @@ protected function getListQuery()
...
+					return (int) $data;
 				},
 				$condition
 			);
-			$condition = implode(',', $condition);
 
 			// Category has to be published
-			$query->where('c.published = 1 AND ws.condition IN (' . $condition . ')');
+			$query->where($db->quoteName('c.published') . ' = 1')
+				->whereIn($db->quoteName('ws.condition') . $condition);
```
Comment (Quy) [inline components/com_content/Model/ArticlesModel.php]:
```suggestion
				->whereIn($db->quoteName('ws.condition'), $condition);
```

File: components/com_content/Model/ArticlesModel.php [hunk 14]
```diff
@@ -519,7 +544,8 @@ function ($data) use ($db)
 					break;
 
 				case 'hits':
-					$query->where('a.hits >= ' . $hitsFilter . ' ');
+					$query->where($db->quoteName('a.hits') .' >= :hits ')
```
Comment (Quy) [inline components/com_content/Model/ArticlesModel.php]:
```suggestion
					$query->where($db->quoteName('a.hits') .' >= :hits')
```

File: components/com_content/Model/ArticleModel.php (hunk 12 above)
Comment (alikon) [reply components/com_content/Model/ArticleModel.php:184]: yes i wasn't able to understand what's the meaning of that `if`, and plus testing work well without
any help ?

File: administrator/components/com_content/Model/FeaturedModel.php [hunk 15]
```diff
@@ -249,29 +258,43 @@ protected function getListQuery()
...
 			{
-				$search = $db->quote('%' . str_replace(' ', '%', $db->escape(trim($search), true) . '%'));
-				$query->where('(a.title LIKE ' . $search . ' OR a.alias LIKE ' . $search . ' OR a.note LIKE ' . $search . ')');
+				$search = '%' . trim($search) . '%';
+				$query->where($db->quoteName('a.title') . ' LIKE :title')
+					->orWhere($db->quoteName('a.alias') . ' LIKE :alias')
+					->orWhere($db->quoteName('a.note') . ' LIKE :note')
+					->bind(':title', $search)
+					->bind(':alias', $search)
+					->bind(':note', $search);
```
Comment (Quy) [inline administrator/components/com_content/Model/FeaturedModel.php]:
```suggestion
					->bind(':note', $search);
```

**Comment by HLeithner** [comment]
@alikon it would be cool if can bring this PR back.

**Comment by alikon** [comment]
re-opened as requested, but it will conflict with #25760

**Comment by alikon** [comment]
sorry but too much changes since it was opened,or i'm a git dumb, but ,i cannot  manage it,  better to start from scratch with a new one

**Pull request closed without merging**